core/pipeline/label.go

```diff
@@ -139,12 +140,31 @@ func LoadGitLabels(workdir string) ([]Label, error) {
 		return nil, err
 	}
 
+	// Check if the commit is a merge commit in the context of PR / MR
+	if (os.Getenv("GITHUB_EVENT_NAME") == "pull_request" ||
+		os.Getenv("CI_PIPELINE_SOURCE") == "merge_request_event") &&
+		commit.NumParents() > 1 {
+		// Get the parent commits
+		parents, err := commit.Parents().Next()
+		if err != nil {
+			return nil, err
+		}
+
+		// Skip the first parent and use the second parent as the previous commit
+		parents, err = parents.Parents().Next()
+		if err != nil {
+			return nil, err
+		}
+
+		commit = parents
```

Comment:
nit: call `commit.Parent(n)` to simplify?

observation: I'm surprised that it's the 2nd parent that has the actual commit we're looking for. Github and Gitlab create two commits per PR?